### ISSUE TITLE
galaxy[tags] does not align between ansible-lint and galaxy-importer

### ISSUE THREAD

**Issue by minsis** [ansible/ansible-lint#4409]
##### Summary

The galaxy-importer library has some additional regex checks on the tags attribute that ansible-lint is missing.

In galaxy-importer you can see a regex is defined for tags where ansible-lint does not:
https://github.com/ansible/galaxy-importer/blob/master/galaxy_importer/constants.py#L47

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.9.0 using ansible-core:2.17.5 ansible-compat:24.9.1 ruamel-yaml:0.18.6 ruamel-yaml-clib:None
A new release of ansible-lint is available: 24.9.0 → 24.10.0 Upgrade by running: pip install --upgrade ansible-lint
```

- ansible installation method: conda
- ansible-lint installation method: conda

##### STEPS TO REPRODUCE

Create basic galaxy.yml file:
```yaml
name: test
namespace: test
description: test
repository: https://repo.com
authors:
  - me
readme: README.md
version: 1.0.0
tags:
  - application
  - tag1
  - tag2
  - tag-3
```

And run `ansible-lint galaxy.yml`

```console (paste below)
ansible-lint galaxy.yml
```

However when importing into galaxy there is a failure on `tag-3`

##### Desired Behavior

ansible-lint should fail on `tag-3`

##### Actual Behavior

ansible-lint passes validation

```paste below
Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
A new release of ansible-lint is available: 24.9.0 → 24.10.0 Upgrade by running: pip install --upgrade ansible-lint
```

**Comment by alisonlhart** [comment]
@minsis Thanks for opening this! We've added this to our roadmap. The implementation should be very straightforward, so if you or anyone would like to open a patch for it, we're happy to review.

**Comment by minsis** [comment]
I want to add that `ansible-lint` also does not validate the max number of allowed tags as well. I'm adding this to my patch for the formatting validation as well.

**Comment by minsis** [comment]
@alisonlhart I wanted to get an opinion on adding some addtional validations to the `galaxy.yml` schema file. I guess overall it's kind of confusing in the way the JSON schema vs the "custom" validations.

Since JSON schema support some of the additional work I'm adding I'm wondering if I should add it there. When it's added my tests are failing on both `schema[galaxy]` and my new validations.
